### ISSUE TITLE
error when trying to compile the example on ESP8266

### ISSUE THREAD

**Issue by AlainPilon** [HackerInside0/Arduino_SharpIR#2]
When compiling the example, on a Acrobotic ESP8266, I am getting this:

```
Archiving built core (caching) in: /var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_cache_849462/core/core_esp8266_esp8266_nodemcuv2_CpuFrequency_80,UploadSpeed_115200,FlashSize_4M3M_eb3c9dcd09e13a6e986604fe76330981.a
Linking everything together...
"/Users/ap/Library/Arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/1.20.0-26-gb404fb9-2/bin/xtensa-lx106-elf-gcc" -g -w -Os -nostdlib -Wl,--no-check-sections -u call_user_start -Wl,-static "-L/Users/ap/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/tools/sdk/lib" "-L/Users/ap/Library/Arduino15/packages/esp8266/hardware/esp8266/2.3.0/tools/sdk/ld" "-Teagle.flash.4m.ld" -Wl,--gc-sections -Wl,-wrap,system_restart_local -Wl,-wrap,register_chipv6_phy  -o "/var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_build_911048/getDistance.ino.elf" -Wl,--start-group "/var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_build_911048/sketch/getDistance.ino.cpp.o" "/var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_build_911048/libraries/SharpIR/SharpIR.a" "/var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_build_911048/arduino.ar" -lm -lgcc -lhal -lphy -lpp -lnet80211 -lwpa -lcrypto -lmain -lwps -laxtls -lsmartconfig -lmesh -lwpa2 -llwip_gcc -lstdc++ -Wl,--end-group  "-L/var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_build_911048"
xtensa-lx106-elf-gcc: error: /var/folders/yx/1tfm1qc50ggf2j6bf6y1ht2svm79th/T/arduino_build_911048/libraries/SharpIR/SharpIR.a: No such file or directory
Using library SharpIR at version 1.1.0 in folder: /Users/ap/Documents/Arduino/libraries/SharpIR 
exit status 1
Error compiling for board NodeMCU 1.0 (ESP-12E Module).
```

Any idea?

**Comment by qub1750ul** [comment]
It seems like a linking problem of the compiler itself.
Can you give me more information about your development environment (OS, IDE, etc ...)?

**Comment by AlainPilon** [comment]
Sure:
- Mac OSX 10.11.6
- Arduino (latest)
- Acrobotic Esp8266 Esp-12e (https://www.amazon.ca/gp/product/B0179SW31O/ref=oh_aui_detailpage_o01_s00?ie=UTF8&psc=1)

I did a few tutorial with the ESP using libraries (to make a simple web server) and it worked so I doubt the issue is there.

I deleted the library folder and reinstalled it and am I still having the same issue.

**Comment by qub1750ul** [comment]
Can you tell me how you have enabled the Arduino IDE to compile for ESP8266, so I can do the tests on my own?

**Comment by AlainPilon** [comment]
Sure. I will reset/reinstall everything tonight and list the steps as I
redo them.

Thanks for the fast response!

**Comment by AlainPilon** [comment]
Ok, here is my settings in the tool menu:

![screen shot 2017-05-24 at 9 29 23 pm](https://cloud.githubusercontent.com/assets/37088/26432220/17d1a5e8-40c8-11e7-84e4-146a9d3a6d5e.png)

Then I opened the SHARPIR example sketch and clicked on the verify button.
I think that is all.

**Comment by ckcampbell248** [comment]
Bumping this issue. I'm getting the same thing. Was any progress made to resolve this?

**Comment by qub1750ul** [comment]
Go to the libraries folder, then in the SharpIR folder.
Open the file library.properties and change the value of dot_a_linkage, from true to false.
This should fix the problem.

Other build systems than arduino's builtin seems to be not affected by this problem, so i will take another bit of time to better investigate the cause before publish a new relese.

**Comment by ckcampbell248** [comment]
Thanks! This worked!

**Comment by qub1750ul** [comment]
Fixed in v2.0.0
